build.py

```diff
@@ -188,9 +188,33 @@ def check_nochanges():
     return True
 
 
-def run_tests(race, parallel, timeout, no_vet):
+def run_tests(race, parallel, timeout, no_vet, verbose):
     """Run the Go test suite on binary output.
     """
+    # NOTE: We download deps here because go fmt on go1.17 first downloads missing deps, printing their names and
+    # giving the appearance of files that need reformatted.
+    logging.info("Downloading dependencies...")
+    try:
+        download_cmd = ["go", "mod", "download"]
+        if verbose:
+            download_cmd.append("-x")
+        subprocess.check_output(download_cmd)
+    except subprocess.CalledProcessError as exc:
+        logging.error("Downloading dependencies failed, see logs for details")
+        logging.error("{}".format(exc.output))
+    # NOTE: We vendor deps here because TestPipelineImplemented in pipeline/tick/tick_test.go uses `importer.For` to
+    # load a few packages from source, and it recursively descends through dependencies. It's apparently not module-
+    # aware and only looks under $GOPATH/src and vendor/.
```

Comment:
Looks like `ForCompiler` in that package is module-aware - can we have that added to the comment here so when we fix it we know where to look?